### ISSUE TITLE
Error: Building package ompl[app]:x86-windows failed with: BUILD_FAILED, when installing OMPL

### ISSUE THREAD

**Issue by ChaochaoLikeNoodles** [microsoft/vcpkg#10753]
Hi, I am new to githug and also new to vcpkg and ompl, I am trying to install ompl on my computer, but got some BUILD_FAILED error, and have no idea why. Thank you for any help.

**Describe the bug**
When installing OMPL on Windows, I always get the Error: Building package ompl:x86-windows failed with: BUILD_FAILED error. 
Log files show:
LINK : fatal error LNK1104: cannot open file 'Eigen3::Eigen.lib'
ninja: build stopped: subcommand failed.
But I already installed Eigen3, when running 
C:\vcpkg_source\vcpkg>vcpkg install eigen3
Computing installation plan...
The following packages are already installed:
    eigen3[core]:x86-windows
Package eigen3:x86-windows is already installed

Total elapsed time: 456.9 us

The package eigen3:x86-windows provides CMake targets:

    find_package(Eigen3 CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Eigen3::Eigen)

**Environment**
- OS: [e.g. Windows/Linux etc...] Windows
- Compiler: revision 
Visual Studio 2019
Vcpkg version: 2020.02.04-nohash

**To Reproduce**
C:\vcpkg_source\vcpkg>vcpkg install ompl[app]
Computing installation plan...
The following packages will be built and installed:
    ompl[app,core]:x86-windows
Starting package 1/1: ompl:x86-windows
Building package ompl[app,core]:x86-windows...
-- Using cached C:/vcpkg_source/vcpkg/downloads/omplapp-1.4.2.tar.gz
-- Using source at C:/vcpkg_source/vcpkg/buildtrees/ompl/src/1.4.2-bf0cc37788
-- Configuring x86-windows
-- Building x86-windows-dbg
-- Restarting Build without parallelism because memory exceeded
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: C:/vcpkg_source/vcpkg/downloads/tools/cmake-3.14.0-windows/cmake-3.14.0-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory: C:/vcpkg_source/vcpkg/buildtrees/ompl/x86-windows-dbg
    See logs for more information:
      C:\vcpkg_source\vcpkg\buildtrees\ompl\install-x86-windows-dbg-out.log
      C:\vcpkg_source\vcpkg\buildtrees\ompl\install-x86-windows-dbg-out-1.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/ompl/portfile.cmake:37 (vcpkg_install_cmake)
  scripts/ports.cmake:90 (include)

Error: Building package ompl:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: ompl:x86-windows
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above.

**Expected behavior**
OMPL should be already built and installed.

**Failure logs** 
-(please attached failure logs)
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4452930/install-x86-windows-dbg-out.log)
[install-x86-windows-dbg-out-1.log](https://github.com/microsoft/vcpkg/files/4452935/install-x86-windows-dbg-out-1.log)

**Additional context**

**Comment by ChaochaoLikeNoodles** [comment]
I was able to successfully run C:\vcpkg_source\vcpkg>vcpkg install ompl, to install the ompl only, not the ompl.app(It failed if I run C:\vcpkg_source\vcpkg>vcpkg install ompl[app]), looks like something wrong caused by the ompl.app, which provides some other library and GUI. I hope this is helpful.

**Comment by cas-bah** [comment]
@ChaochaoLikeNoodles Hi, I also got

> LINK : fatal error LNK1104: cannot open file 'Eigen3::Eigen.lib'
> ninja: build stopped: subcommand failed.

when running the ompl[app] installation. Did you ever get it to work?

**Comment by LilyWangL** [comment]
Thanks for posting this issue. This issue has been fixed. You can update VCPKG and rebuild port. So I am closing this issue for now.

**Comment by cas-bah** [comment]
Hi, I just installed VCPKG on another machine.

 .\vcpkg install ompl[app]

leads to the following message:

Computing installation plan...
The following packages will be built and installed:
    ompl[app,core]:x86-windows
Starting package 1/1: ompl:x86-windows
Building package ompl[app,core]:x86-windows...
-- Using cached C:/vcpkg/downloads/omplapp-1.4.2.tar.gz
-- Extracting source C:/vcpkg/downloads/omplapp-1.4.2.tar.gz
-- Applying patch fix-findeigen3.patch
CMake Error at scripts/cmake/vcpkg_apply_patches.cmake:51 (message):
  Applying patch failed.  Checking patch CMakeLists.txt...

  error: while searching for:

  set_package_properties(Eigen3 PROPERTIES

  URL "http://eigen.tuxfamily.org"
  PURPOSE "A linear algebra library used throughout OMPL.")

  find_package(Eigen3 REQUIRED)

  include_directories("${EIGEN3_INCLUDE_DIR}")



  set_package_properties(MORSE PROPERTIES



  error: patch failed: CMakeLists.txt:110

  error: CMakeLists.txt: patch does not apply

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:134 (vcpkg_apply_patches)
  ports/ompl/portfile.cmake:18 (vcpkg_extract_source_archive_ex)
  scripts/ports.cmake:90 (include)


Error: Building package ompl:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: ompl:x86-windows
  Vcpkg version: 2020.02.04-nohash

**Comment by ChaochaoLikeNoodles** [comment]
I also tried again on my computer and got the same error message as @cas-bah mentioned in the comment on April 22, 2020.

**Comment by Petrel1i** [comment]
Hi, I also got the same error message as @cas-bah mentioned

**Comment by gautz** [comment]
Same here with ``vcpkg install ompl[app]:x64-windows`` which equals ``ompl[app,core]:x64-windows``

Here is the log for the build
```
Starting package 100/100: ompl:x64-windows
Building package ompl[app,core]:x64-windows...
-- Downloading https://bitbucket.org/ompl/ompl/downloads/omplapp-1.4.2-Source.t
-- Extracting source D:/vcpkg/downloads/omplapp-1.4.2.tar.gz
-- Applying patch fix-findeigen3.patch
CMake Error at scripts/cmake/vcpkg_apply_patches.cmake:51 (message):
  Applying patch failed.  Checking patch CMakeLists.txt...

  error: while searching for:

  set_package_properties(Eigen3 PROPERTIES

      URL "http://eigen.tuxfamily.org"
      PURPOSE "A linear algebra library used throughout OMPL.")

  find_package(Eigen3 REQUIRED)

  include_directories("${EIGEN3_INCLUDE_DIR}")



  set_package_properties(MORSE PROPERTIES



  error: patch failed: CMakeLists.txt:110

  error: CMakeLists.txt: patch does not apply

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:134 (vcpkg_apply_patches)
  ports/ompl/portfile.cmake:18 (vcpkg_extract_source_archive_ex)
  scripts/ports.cmake:90 (include)


Error: Building package ompl:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: ompl:x64-windows
  Vcpkg version: 2020.02.04-nohash
```
Interestingly ``ompl:x64-windows`` which equals ``ompl[core]:x64-windows`` is building properly

**Comment by timautry** [comment]
I also get this same issue on x64-windows and eigen3 is installed:
->`vcpkg install ompl[app,core]`
Computing installation plan...
The following packages will be built and installed:
    ompl[app,core]:x64-windows
Starting package 1/1: ompl:x64-windows
Building package ompl[app,core]:x64-windows...
-- Downloading https://bitbucket.org/ompl/ompl/downloads/omplapp-1.4.2-Source.tar.gz...
-- Extracting source D:/Git/vcpkg.downloads/omplapp-1.4.2.tar.gz
-- Applying patch fix-findeigen3.patch
CMake Error at scripts/cmake/vcpkg_apply_patches.cmake:51 (message):
  Applying patch failed.  Checking patch CMakeLists.txt...

  error: while searching for:

  set_package_properties(Eigen3 PROPERTIES

      URL "http://eigen.tuxfamily.org"
      PURPOSE "A linear algebra library used throughout OMPL.")

  find_package(Eigen3 REQUIRED)

  include_directories("${EIGEN3_INCLUDE_DIR}")

  set_package_properties(MORSE PROPERTIES

  error: patch failed: CMakeLists.txt:110

  error: CMakeLists.txt: patch does not apply

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:134 (vcpkg_apply_patches)
  ports/ompl/portfile.cmake:18 (vcpkg_extract_source_archive_ex)
  scripts/ports.cmake:90 (include)

Error: Building package ompl:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: ompl:x64-windows
  Vcpkg version: 2020.02.04-nohash-external

Additionally, attach any relevant sections from the log files above.

SYSTEM@ANDROMEDA 05/09/2020 21:03:00 D:\Git\vs2019.prod
->`vcpkg list eigen`
ceres[eigensparse]:x64-windows                                      Use of Eigen as a sparse linear algebra library ...
eigen3:x64-windows                                 3.3.7-4          C++ template library for linear algebra: matrice...

**Comment by LilyWangL** [comment]
This issue has been fixed by PR #10972. Please update VCPKG and rebuild this port.
